### ISSUE TITLE
Programatically ask for permission to send push notifs

### ISSUE THREAD

**Issue by gwendall** [raix/push#93]
First off, thanks for this great package.

From what I have seen, the apps using this package ask for the user permission to receive push notifications right away, on startup. 

It does not seem to be the best pattern - see [this interesting article](http://techcrunch.com/2014/04/04/the-right-way-to-ask-users-for-ios-permissions/). 

Is there a way we could control when to ask for the permission?

**Comment by raix** [comment]
it depends on the cordova push plugin https://github.com/raix/PushPlugin (a cloned working version) - If wanting to change the behaviour that would be the place

**Comment by gwendall** [comment]
Thanks the the answer @raix. For those interested in this question, see https://github.com/phonegap-build/PushPlugin/issues/378.

**Comment by mikepaszkiewicz** [comment]
@gwendall I followed up on that issue, the most recent Phonegap push plugin supports delaying push: https://github.com/phonegap/phonegap-plugin-push

**Comment by derwaldgeist** [comment]
+1 I also think that asking for push notifications on startup is not a good idea, especially if the user does not know much about the app's purpose yet. It would be great if the push delay could be implemented.

**Comment by derwaldgeist** [comment]
I'm getting quite bad feedback on my app due to this. Any chance that this will be changed soon? The Cordova plugin supports this now, so this should not be too complicated. Would be great, thanks!

**Comment by mikepaszkiewicz** [comment]
@raix With 1.3 supporting cordova-ios 4.0.1, this looks to be a lot more feasible?
[This issue](https://github.com/phonegap-build/PushPlugin/issues/378) implies changing the push plugin dep version to 1.5.2 would enable programmatic control of the register event - in addition to changing some behavior of the Configure method on class PushHandler (lib/client/cordova.js)

I've only recently started poking around the package internals, and might start to mess around with the behavior. curious to get your thoughts on sizing up the task at hand in the meantime?

**Comment by raix** [comment]
@mikepaszkiewicz yeah - It seems like I have to upgrade the cordova plugin, might add an option for manually requesting push access

**Comment by derwaldgeist** [comment]
That would be awesome! :+1:

**Comment by mikepaszkiewicz** [comment]
by upgrading the cordova plugin - are there any changes to actual package code, or just bumping the Cordova.require to 1.5.2?

**Comment by raix** [comment]
@mikepaszkiewicz I havent investigated - code wise I'd probably just bump cordova, clean up things and add a couple of features

-

I'm considering writing a QA test plan so others can help out QA things - it might speed up development - What do you guys think, would that be feasible?

**Comment by mikepaszkiewicz** [comment]
All for it, a QA test plan would be an awesome place to start out. Can definitely contribute where needed!

 		
Mike Paszkiewicz
Cofounder & COO, Habitat LLC
443-216-9367 <tel:443-216-9367> | mike@tryhabitat.com <mailto:mike@tryhabitat.com> | tryhabitat.com <http://tryhabitat.com/> | Philadelphia, PA
 <http://facebook.com/mikepaszkiewicz>  <http://www.linkedin.com/in/michael-paszkiewicz-07053283>

**Comment by markoshust** [comment]
Just downloaded this plugin and trying to get the final touches working. However, this is a big issue. I wanted to know before I went and dug deep on this, has progress been made on this?

**Comment by markoshust** [comment]
fyi @raix it appears this issue can now be easily solved because of https://github.com/raix/push/issues/252#issuecomment-255372516

do you know what exactly would be involved in implementing? is it modifying this line https://github.com/raix/push/blob/524f23ac3c3c968a237bfbbf1b77f324d5c830bc/lib/client/cordova.js#L61 to work from options defined instead of startup? i don't mind taking this task unless this has been already implemented.

**Comment by raix** [comment]
I agree - we should give more control,

**Comment by jensljungblad** [comment]
Also interested in this! Any updates?
